tests/test_hub_service.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+from flask import session
+from jupyter_smart_on_fhir.auth import get_jwks_from_key
+from jupyter_smart_on_fhir.hub_service import (
+    create_app,
+    set_encrypted_cookie,
+    get_encrypted_cookie,
+    prefix,
+    token_for_code,
+)
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+import os
+import requests
+from conftest import SandboxConfig
+from urllib import parse
+
+
+@pytest.fixture(scope="module")
+def keys(tmp_path_factory):
+    tmp_path = tmp_path_factory.mktemp("keys")
+    private_key = rsa.generate_private_key(public_exponent=65537, key_size=4096)
+
+    private_pem = private_key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
+    public_pem = private_key.public_key().public_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PublicFormat.SubjectPublicKeyInfo,
+    )
+
+    key_name = "jwtRS256.key"
+    private_key_path = tmp_path / key_name
+    public_key_path = tmp_path / f"{key_name}.pub"
+    private_key_path.write_bytes(private_pem)
+    public_key_path.write_bytes(public_pem)
+    return {"SSH_KEY_PATH": str(private_key_path), "SSH_KEY_ID": "test_key"}
+
+
+@pytest.fixture(scope="function")
```

Comment:
function is the dedfault scope, you can use:

```suggestion
@pytest.fixture
```

for most fixtures